tests/metric_providers/test_elephant_providers.py

```
@@ -0,0 +1,260 @@
+import requests
+import uuid
+from datetime import datetime, timezone
+
+
+from pathlib import Path
+from unittest.mock import patch, MagicMock
+
+from metric_providers.carbon.intensity.elephant.machine.provider import CarbonIntensityElephantMachineProvider
+from metric_providers.base import MetricProviderConfigurationError
+
+GMT_METRICS_DIR = Path('/tmp/green-metrics-tool/metrics')
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/green-metrics-tool/metrics"

(S108)